model_analyzer/config/generate/model_variant_name_manager.py

```diff
@@ -21,95 +22,78 @@ class ModelVariantNameManager:
-            model_name_index = self._get_next_model_name_index(base_model_name)
-            variant_name = f'{base_model_name}_config_{model_name_index}'
-            self._add_model_variant_name(base_model_name, param_combo,
-                                         variant_name)
-        return variant_name
-
-    def _add_model_variant_name(self, base_model_name, param_combo,
-                                variant_name):
-        """ 
-        Save a new model variant name for this base model
-        """
-        if base_model_name not in self._variant_names:
-            self._variant_names[base_model_name] = {}
+        model_variant_name = self._create_new_model_variant(model_name, mcd)
 
-        hashable_key = self._get_hashable_key(param_combo)
-        self._variant_names[base_model_name][hashable_key] = variant_name
+        return (False, model_variant_name)
 
-    def _get_next_model_name_index(self, base_model_name):
-        """
-        Get the next index to be used as a unique number for variant naming
-        """
-        if base_model_name not in self._model_name_index:
-            self._model_name_index[base_model_name] = 0
-        else:
-            self._model_name_index[base_model_name] += 1
+    def _copy_and_restore_mcd_name(self, model_name: str,
+                                   model_config_dict: Dict) -> Dict:
+        mcd = deepcopy(model_config_dict)
+        mcd['name'] = model_name
 
-        return self._model_name_index[base_model_name]
+        return mcd
 
-    def _get_hashable_key(self, obj):
-        obj_copy = deepcopy(obj)
+    def _find_existing_variant(self,
+                               model_config_dict: Dict) -> Tuple[bool, str]:
+        for model_config_name, model_config_variant_dict in self._model_config_dicts.items(
+        ):
+            if model_config_dict == model_config_variant_dict:
+                return (True, model_config_name)
 
-        return self._get_hashable_key_helper(obj_copy)
+        return (False, "")
 
-    def _get_hashable_key_helper(self, obj):
-        """
-        Given a list or dict which may have nested dicts/lists, return a key that 
-        is hashable 
-        """
+    def _is_default_config(self, param_combo: Dict) -> bool:
+        return param_combo == DEFAULT_CONFIG_PARAMS
 
-        if isinstance(obj, collections.abc.Hashable):
-            key = obj
-        elif isinstance(obj, collections.abc.Mapping):
-            key = frozenset(
-                (k, self._get_hashable_key_helper(v)) for k, v in obj.items())
-        elif isinstance(obj, collections.abc.Iterable):
-            key = tuple(self._get_hashable_key_helper(item) for item in obj)
+    def _create_new_model_variant(self, model_name: str,
+                                  model_config_dict: Dict) -> str:
+        if model_name not in self._model_name_index:
+            self._model_name_index[model_name] = 0
+            model_config_name = model_name + '_config_0'
```

Comment:
combine some of the if/else.  Only the determining of new_index and incrementing of model_name_index{} needs to be in the if/else

Reply:
Done